DOCSP-37065 - v2.24 updates (#173)
mongodb/docs-csharp: source/whats-new.txt

.. _upcoming-breaking-changes:

.. _version-2.24:

What's New in 2.24
------------------

The 2.24 driver release includes the following new features:

> - Added `IMongoCollection.DistinctMany() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.IMongoCollection-1.DistinctMany.html>`__ method.

Reviewer: S: Since the last few bullets in this list are complete sentences, I think it makes sense to update the ones that are sentence fragments

```suggestion
- Added the `IMongoCollection.DistinctMany() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.IMongoCollection-1.DistinctMany.html>`__ method.
```

Reviewer: Applies to the other bullet points too

- Added support for latest optional arguments for :manual:`$dateFromString </reference/operator/aggregation/dateFromString/>`
  aggregation expression.

> - Added support for sorting by score in :manual:`$search </reference/operator/aggregation/search/>`.

Reviewer: I: extra period here
```suggestion
- Added support for sorting by score in :manual:`$search </reference/operator/aggregation/search/>`
```

  aggregation stage.
- Added ``VectorSearchScore`` builder for :manual:`$vectorSearch </reference/operator/aggregation/vectorSearch/>`
  aggregation stage.
- Added support for `$in <https://www.mongodb.com/docs/manual/reference/operator/query/in/>`__
  operator in Atlas Search.
- Added ``Mql.Exists()``, ``Mql.IsMissing()``, and ``Mql.IsNullOrMissing()`` methods for
  LINQ queries.
- Added support for ``IComparable.CompareTo()`` method in LINQ queries.
- Added support for nested calls to ``AsQueryable()`` method to support LINQ queries
  by the `MongoDB Entity Framework Core Provider <https://www.mongodb.com/docs/entity-framework/current/>`__.
- Added support for TLS 1.3.
- Updated ``libmongocrypt`` package version to 1.8.2.
- Fixed segfault in Kerberos (``libgssapi``) on newer Linux distributions.

> - Fixed a GridFS error when attempting to upload a GridFS file with a duplicate ID where
>   the new file is smaller than the original file.

Reviewer: S: the wording is a little confusing here, maybe you could say something like this (based on [CSHARP-4900](https://jira.mongodb.org/browse/CSHARP-4900)):

```suggestion
- Fixed a GridFS error in which uploading a duplicate GridFS file larger than the original file caused errors when downloading the original file
```
